libs/block-content/website/src/lib/poll/poll-block.provider.tsx

```diff
@@ -33,6 +36,29 @@ export function PollBlockProvider({children}: PropsWithChildren) {
     [settings?.settings]
   )
 
+  async function vote(
```

Comment:
I am a bit conflicted about this function as IMO, I understand **why** this was created.

It not doing anything when a user can not vote is problematic design as it's impure in this case, but it's inherently impure because it calls an API. It also causes a double check if a user can vote as the UI already does this.

Double checking is a problem because the UI & this function may get out of sync, for example this function isn't correct as it tries to vote on polls that are closed unlike the UI that does check for that.

Imo the canVote probably should be passed down, and this function should **not** safeguard, but the one calling vote should. It's essentially a 

```
if (canVote) {
  await vote(...)
}
```

in the useEffect but not in the UI as that already safeguards against it by not showing the buttons. I think that's a lot cleaner in design and fixes there not being a single source of truth if a user can vote or not.